src/app/core/store/tables/tables.effects.ts

```diff
@@ -600,43 +602,63 @@ export class TablesEffects {
         first(),
         filter(([table]) => !!table),
         mergeMap(([table, documents, moveCursorDown]) => {
-          const {cursor} = action.payload;
-          const {rows} = table.config;
-
-          const createdDocuments = filterNewlyCreatedDocuments(rows, documents);
-          const unknownDocuments = filterUnknownDocuments(rows, documents);
-
-          const actions: Action[] = [];
-
-          if (createdDocuments.length > 0) {
-            actions.push(
-              new TablesAction.InitRows({
-                cursor: {...cursor, rowPath: []},
-                documents: createdDocuments,
-                linkInstances: [],
-              })
-            );
-            if (moveCursorDown) {
-              actions.push(new TablesAction.MoveCursor({direction: Direction.Down}));
-            }
-          }
+          const {collectionId} = table.config.parts[0];
+          return this.store$.pipe(
+            select(selectCollectionById(collectionId)),
+            take(1),
+            mergeMap(collection => this.collectionPermissionsPipe.transform(collection)),
+            take(1),
+            map(permissions => permissions.write || permissions.writeWithView),
```

Comment:
It's enough to test if user has `writeWithView` permission, but you can leave it as it is.

Reply:
I would leave it here so in the future we can remove the logic for adding empty rows elsewhere.